site/_docs/reference.md

```diff
@@ -2806,14 +2806,14 @@ BigQuery's type system uses confusingly different names for types and functions:
 | h s | string1 NOT RLIKE string2                    | Whether *string1* does not match regex pattern *string2* (similar to `NOT LIKE`, but uses Java regex)
 | b o | RPAD(string, length[, pattern ])             | Returns a string or bytes value that consists of *string* appended to *length* with *pattern*
 | b o | RTRIM(string)                                | Returns *string* with all blanks removed from the end
-| b | SAFE_ADD(numeric1, numeric2)                   | Returns *numeric1* + *numeric2*, or NULL on overflow
+| b | SAFE_ADD(numeric1, numeric2)                   | Returns *numeric1* + *numeric2*, or NULL on overflow.  Only supports BIGINT, DOUBLE, or DECIMAL arguments
```

Comment:
In fact, you want to emphasize that It will not throw an error when it exceeds integer.

But for me, the comment is not helpful.

Reply:
The fact that it does not throw was already in the message.
I want to emphasize that it does not work as you may expect for integers.

Reply:
I think there will be ambiguity. The newly added comment will make users think that this function can only use BIGINT, DOUBLE, or DECIMAL. The internal transformation of integer is only internal processing.

Reply:
But this is exactly how all SQL functions work. For example, trigonometric functions are only defined for DOUBLE values. When you call them the compiler inserts an implicit cast for the argument to convert it to double. This kind of "internal processing" happens almost always, it has nothing to do with these functions. It is really part of the definition of SQL. These functions indeed can only use the types you enumerate, if you use a value with another type as argument it will be cast to one of these three types. That is exactly what I am documenting.

What is important in this case is to understand that one cannot use SAFE_* to perform safe computations on integers. If you do SAFE_ADD(int, int) the result will be a BIGINT. If you try to assign this to an integer column and it overflows you will get an exception, which is completely unexpected since this is what the SAFE function was designed to avoid. You also need a SAFE_CAST if you want to avoid the error.